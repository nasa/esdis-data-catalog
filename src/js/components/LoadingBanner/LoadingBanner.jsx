import React from 'react'
import PropTypes from 'prop-types'
import Row from 'react-bootstrap/Row'
import Spinner from 'react-bootstrap/Spinner'

export const LoadingBanner = ({
  dataTestId
}) => (
  <Row className="justify-content-center mt-5">
    <Spinner animation="grow" variant="primary" data-testid={dataTestId} />
  </Row>
)

LoadingBanner.defaultProps = {
  dataTestId: 'loading-banner__spinner'
}

LoadingBanner.propTypes = {
  dataTestId: PropTypes.string
}

export default LoadingBanner
