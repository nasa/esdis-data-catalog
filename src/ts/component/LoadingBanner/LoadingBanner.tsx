import React from 'react'

import Row from 'react-bootstrap/Row'
import Spinner from 'react-bootstrap/Spinner'

interface LoadingBannerProps {
  dataTestId?: string;
}

export const LoadingBanner: React.FC<LoadingBannerProps> = ({
  dataTestId = 'loading-banner__spinner'

}) => (
  <Row className="justify-content-center mt-5">
    <Spinner animation="grow" variant="primary" data-testid={dataTestId} />
  </Row>
)

export default LoadingBanner
