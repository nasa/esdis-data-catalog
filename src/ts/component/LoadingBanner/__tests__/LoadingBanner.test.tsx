import React from 'react'

import { render } from '@testing-library/react'

import { LoadingBanner } from '../LoadingBanner'

describe('Error Banner Component', () => {
  describe('Provided a message', () => {
    test('it renders a loading banner', () => {
      const { container } = render(
        <LoadingBanner />
      )

      // Match the snapshot
      expect(container).toMatchSnapshot()
    })
  })

  describe('Provided a test id', () => {
    test('it renders a loading banner', () => {
      const { container } = render(
        <LoadingBanner dataTestId="test-spinner" />
      )

      // Match the snapshot
      expect(container).toMatchSnapshot()
    })
  })
})
