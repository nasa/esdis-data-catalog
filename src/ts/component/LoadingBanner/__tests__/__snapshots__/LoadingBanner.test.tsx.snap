// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error Banner Component > Provided a message > it renders a loading banner 1`] = `
<div>
  <div
    class="justify-content-center mt-5 row"
  >
    <div
      class="spinner-grow text-primary"
      data-testid="loading-banner__spinner"
    />
  </div>
</div>
`;

exports[`Error Banner Component > Provided a test id > it renders a loading banner 1`] = `
<div>
  <div
    class="justify-content-center mt-5 row"
  >
    <div
      class="spinner-grow text-primary"
      data-testid="test-spinner"
    />
  </div>
</div>
`;
